iocore/net/quic/QUICFrame.cc

```diff
@@ -2098,13 +2104,7 @@ QUICFrameFactory::create_stream_frame(const uint8_t *data, size_t data_len, QUIC
 QUICFrameUPtr
 QUICFrameFactory::split_frame(QUICFrame *frame, size_t size)
 {
-  if (frame->type() != QUICFrameType::STREAM) {
-    return QUICFrameFactory::create_null_frame();
-  }
-
-  QUICStreamFrame *old_frame = static_cast<QUICStreamFrame *>(frame);
-  QUICStreamFrame *new_frame = old_frame->split(size);
-  return QUICFrameUPtr(new_frame, &QUICFrameDeleter::delete_stream_frame);
+  return QUICFrameUPtr(frame->split(size), &QUICFrameDeleter::delete_stream_frame);
```

Comment:
We should probably add null check here or in delete_*_frame. Otherwise nullptr well be dereferenced in delete_*_frame.